app/components/Menu/MenuDropdownItem.tsx

```diff
@@ -0,0 +1,131 @@
+import { Pressable, Text, View, type ViewStyle, type TextStyle } from "react-native"
+import { useState, useRef, memo, useCallback } from "react"
+import { themed } from "../../theme/theme"
+import { menuSettings } from "./menuSettings"
+import type { MenuItem } from "./types"
+
+interface MenuDropdownItemProps {
+  item: MenuItem
+  index: number
+  onItemPress: (item: MenuItem) => void
+  onItemHover: (itemLabel: string, index: number, hasSubmenu: boolean) => void
+}
+
+const MenuDropdownItemComponent = ({
+  item,
+  index,
+  onItemPress,
+  onItemHover,
+}: MenuDropdownItemProps) => {
+  const [hoveredItem, setHoveredItem] = useState<string | null>(null)
+  const hoverTimeoutRef = useRef<NodeJS.Timeout | null>(null)
+  const enabled = item.enabled !== false
+
+  const handleHoverIn = useCallback(() => {
+    // Clear any pending hover clear
+    if (hoverTimeoutRef.current) {
+      clearTimeout(hoverTimeoutRef.current)
+      hoverTimeoutRef.current = null
+    }
+    setHoveredItem(item.label)
+    const hasSubmenu = !!item.submenu
+    onItemHover(item.label, index, hasSubmenu)
+  }, [item.label, item.submenu, index, onItemHover])
+
+  const handleHoverOut = useCallback(() => {
+    // Use a small timeout to prevent flickering between items
+    hoverTimeoutRef.current = setTimeout(() => {
+      setHoveredItem((current) => (current === item.label ? null : current))
+    }, 10)
+  }, [item.label])
+
+  const handlePress = useCallback(() => {
+    if (!item.action || !enabled) return
+    item.action()
+    onItemPress(item)
+  }, [item, onItemPress])
+
+  return (
+    <Pressable
+      onHoverIn={handleHoverIn}
+      onHoverOut={handleHoverOut}
+      onPress={handlePress}
+      disabled={!enabled}
```

Comment:
This gives more weight to the idea that we might want to use `disabled` instead of `enabled`.